core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -157,6 +161,14 @@ class HistoryServer(
     contextHandler.setContextPath(HistoryServer.UI_PATH_PREFIX)
     contextHandler.addServlet(new ServletHolder(loaderServlet), "/*")
     attachHandler(contextHandler)
+
+    // Register history server source to history server metrics system.
+    cacheMetrics.init()
+    metricsSystem.registerSource(provider.getHistoryServerSource())
+    metricsSystem.registerSource(cacheMetrics)
+    metricsSystem.start(false)
```

Comment:
can we add a comment about why we are passing false here ?

Reply:
> can we add a comment about why we are passing false here ?

Done